src/uu/shred/src/shred.rs

```diff
@@ -212,136 +213,163 @@ impl<'a> BytesGenerator<'a> {
+    let matches = App::new("shred")
+        .name("shred")
+        .version(VERSION_STR)
+        .about(
+            "Overwrite the specified FILE(s) repeatedly, in order to make it harder\n\
+for even very expensive hardware probing to recover the data.",
+        )
+        .after_help(AFTER_HELP)
+        .usage("shred [OPTION]... FILE...")
+        .arg(Arg::with_name(options::FILE).hidden(true).multiple(true))
+        .arg(
+            Arg::with_name(options::ITERATIONS)
+                .long(options::ITERATIONS)
+                .short("n")
+                .help("overwrite N times instead of the default (3)")
+                .value_name("NUMBER"),
+        )
+        .arg(
+            Arg::with_name(options::SIZE)
+                .long(options::SIZE)
+                .short("s")
+                .takes_value(true)
+                .value_name("N")
+                .help("shred this many bytes (suffixes like K, M, G accepted)"),
+        )
+        .arg(
+            Arg::with_name(options::REMOVE)
+                .short("u")
+                .long(options::REMOVE)
+                .help("truncate and remove file after overwriting;  See below"),
+        )
+        .arg(
+            Arg::with_name(options::VERBOSE)
+                .long(options::VERBOSE)
+                .short("v")
+                .help("show progress"),
+        )
+        .arg(
+            Arg::with_name(options::EXACT)
+                .long(options::EXACT)
+                .short("x")
+                .help(
+                    "do not round file sizes up to the next full block;\n\
+this is the default for non-regular files",
+                ),
+        )
+        .arg(
+            Arg::with_name(options::ZERO)
+                .long(options::ZERO)
+                .short("z")
+                .help("add a final overwrite with zeros to hide shredding"),
+        )
+        .get_matches_from(args);
+
+    let mut errs: Vec<String> = vec![];
+
+    if !matches.is_present(options::FILE) {
+        eprintln!("{}: Missing an argument", NAME);
+        eprintln!("For help, try '{} --help'", NAME);
         return 0;
-    } else {
-        let iterations = match matches.opt_str("iterations") {
-            Some(s) => match s.parse::<usize>() {
-                Ok(u) => u,
-                Err(_) => {
-                    println!("{}: Invalid number of passes", NAME);
-                    return 1;
-                }
-            },
-            None => 3,
-        };
-        let remove = matches.opt_present("remove");
-        let size = get_size(matches.opt_str("size"));
-        let exact = matches.opt_present("exact") && size.is_none(); // if -s is given, ignore -x
-        let zero = matches.opt_present("zero");
-        let verbose = matches.opt_present("verbose");
-        for path_str in matches.free.into_iter() {
-            wipe_file(&path_str, iterations, remove, size, exact, zero, verbose);
+    }
+
+    let mut iterations: usize = 3;
```

Comment:
Why you aren't using the default_values options from clap?

Reply:
Changed, thanks! I am still new to clap, so I missed this feature.